examples/reliable_conversation/src/utils/progress_reporter.py

```
@@ -0,0 +1,145 @@
+        if len(prompt) > truncate_at:
+            truncated_prompt = prompt[:truncate_at] + f"\n[dim]... (truncated, {len(prompt)} total chars)[/dim]"
+        else:
+            truncated_prompt = prompt
+            
+        if len(response) > truncate_at:
+            truncated_response = response[:truncate_at] + f"\n[dim]... (truncated, {len(response)} total chars)[/dim]"
+        else:
+            truncated_response = response
+        
+        self.console.print(f"\n[dim]🤖 {role} LLM Interaction ({elapsed:.1f}s):[/dim]")
+        self.console.print(f"[dim]┌─ Prompt:[/dim]")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)